### ISSUE TITLE
Can't run introduction-to-generic-pipelines, got  invalidPipeline error

### ISSUE THREAD

**Issue by eromoe** [elyra-ai/elyra#2145]
Error message

```
{
  "severity": 1,
  "source": "Elyra Pipeline Validation Service",
  "type": "invalidPipeline",
  "message": "Primary pipeline version field has an invalid value.",
  "data": {
    "supported_version": 4,
    "detected_version": 5
  }
}
```

**Comment by ptitzler** [comment]
I reproduced this using 3.1.0/pip and local build from master

**Comment by ptitzler** [comment]
Although the issue is not caused by it, there is a bug in the [validation logic](https://github.com/elyra-ai/elyra/blob/master/elyra/pipeline/validation.py#L169-L187) that is exposed. In the reported scenario (supported_version < detected_version) the correct code path is never reached and an incorrect validation message produced.

**Comment by eromoe** [comment]
@ptitzler Yes , elyra  is latest  3.1.0 .  
PS: this error doesn't show in  elyra/elyra:dev  docker image

**Comment by ajbozarth** [comment]
This was caused by pinning the minor version of pipeline-editor instead of the patch version. This will be fixed in #2143

**Comment by ptitzler** [comment]
@eromoe if you upgrade to version 3.1.1 things should work as expected.
